modules/dnn/test/test_model.cpp

```diff
@@ -738,6 +914,178 @@ TEST_P(Test_Model, TextDetectionByEAST)
+    gt_boxes.push_back(Rect(408, 84, 34, 49));
+    gt_boxes.push_back(Rect(142, 256, 32, 43));
+    gt_boxes.push_back(Rect(118, 46, 34, 45));
+    gt_boxes.push_back(Rect(340, 34, 29, 42));
+    gt_boxes.push_back(Rect(65, 123, 37, 43));
+    gt_boxes.push_back(Rect(259, 228, 32, 43));
+
+    std::vector<float> gt_confidences;
+    gt_confidences.push_back(0.672344f);
+    gt_confidences.push_back(0.665494f);
+    gt_confidences.push_back(0.646890f);
+    gt_confidences.push_back(0.388749f);
+    gt_confidences.push_back(0.283762f);
+    gt_confidences.push_back(0.256311f);
+
+    // Threshold
+    const double iouDiff = 0.9;
+    const double scoreDiff = 2e-4;
+
+    // Run Test
+    testFaceDetectionModelBySSD(
+        face_detection_weight_path,
+        face_detection_config_path,
+        image_path,
+        gt_boxes,
+        gt_confidences,
+        iouDiff,
+        scoreDiff
+    );
+}
+
+TEST_P(Test_Model, FaceDetectionByYN)
+{
+    if (target == DNN_TARGET_OPENCL_FP16)
+        applyTestTag(CV_TEST_TAG_DNN_SKIP_OPENCL_FP16);
+
+    // Weight
+    const std::string face_detection_weight_path = _tf("onnx/models/yunet-202109.onnx", false);
+
+    // Ground Truth
+    const std::string test_labels = findDataFile("cv/dnn_face/detection/cascades_labels.txt");
+    std::ifstream ifs(test_labels.c_str());
+    CV_Assert(ifs.is_open());
+
+    std::string image_path;
+    {
+        std::string line;
+        getline(ifs, line);
+        std::istringstream iss(line);
+        iss >> image_path;
+    }
+    CV_Assert(!image_path.empty());
+    image_path = findDataFile("cv/cascadeandhog/images/" + image_path);
+
+    uint num_faces = 0;
+    {
+        std::string line;
+        getline(ifs, line);
+        std::istringstream iss(line);
+        iss >> num_faces;
+    }
+    CV_Assert(num_faces != 0);
+
+    std::vector<Rect> gt_boxes;
+    std::vector<std::vector<Point>> gt_landmarks;
+    {
+        for (uint i = 0; i < num_faces; i++)
+        {
+            std::string line, part;
+            getline(ifs, line);
+            std::stringstream ss{ line };
+
+            int box[4];
+            for (int j = 0; j < 4; j++)
+            {
+                getline(ss, part, ' ');
+                std::istringstream iss(part);
+                iss >> box[j];
+            }
+            gt_boxes.push_back(Rect(box[0], box[1], box[2], box[3]));
+
+            std::vector<Point> points;
+            for(uint j = 4; j < 14; j+=2)
+            {
+                int x, y;
+                getline(ss, part, ' ');
+                std::istringstream iss_x(part);
+                iss_x >> x;
+                getline(ss, part, ' ');
+                std::istringstream iss_y(part);
+                iss_y >> y;
+                points.push_back(Point(x, y));
+            }
+            gt_landmarks.push_back(points);
+        }
+
+        // There are 7 faces in the addams-family.png, but yunet-202109.onnx can't detect the last one.
```

Comment:
just a note: all faces found if `score_threshold = 0.85`

Reply:
Oh, It is my mistake. The threshold was not applied correctly. I will fix it. Thanks,

Reply:
fixed it!